modules/core/src/main/java/org/apache/synapse/mediators/builtin/LogMediator.java

```diff
@@ -197,7 +197,9 @@ private String getFullLogMessage(MessageContext synCtx) {
         sb.append(getSimpleLogMessage(synCtx));
         try {
             org.apache.axis2.context.MessageContext a2mc = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
-            if (JsonUtil.hasAJsonPayload(a2mc)) {
+            // If there is a XML element which store sourced JSON payload, synCtx.getEnvelope() will not be null.
+            // So need to check it also to avoid log in JSON format i.e. {"value" : "test"}
+            if (JsonUtil.hasAJsonPayload(a2mc) && synCtx.getEnvelope() == null) {
```

Comment:
Isn't it better to move the condition in else if to top? That way we don't have to add the extra check (synCtx.getEnvelope() == null)).